packages/util/src/extendedAssets/index.ts

```diff
@@ -0,0 +1,135 @@
+import type { Api } from '../api';
+import { Asset } from '../assets/types';
+import { SoulBoundToken } from './types';
+
+export class ExtendedAssetsModule<T> {
+  constructor(private readonly root: Api<T>) {}
+
+  /**
+   * Get SBT meta info by collecting it from different states
+   * @param assetId asset ID
+   *
+   */
+  public async getSbtMetaInfo(assetId: string): Promise<SoulBoundToken> {
+    const sbtSpecificInfo = await this.root.api.query.extendedAssets.soulboundAsset(assetId);
+    const sbtCommonInfo = await this.root.api.query.assets.assetInfosV2(assetId);
+
+    const { externalUrl, issuedAt, regulatedAssets } = sbtSpecificInfo.unwrapOrDefault();
+    const { symbol, name, assetType, contentSource, description } = sbtCommonInfo.toHuman() as Partial<SoulBoundToken>;
+
+    return {
+      address: assetId,
+      symbol,
+      name,
+      assetType,
+      contentSource,
+      description,
+      externalUrl: externalUrl.toHuman() as string,
+      issuedAt: issuedAt.toHuman(),
+      regulatedAssets: (regulatedAssets.toJSON() as Array<{ code: string }>).map(
+        (regulatedAsset) => regulatedAsset?.code
+      ),
+    };
+  }
+
+  /**
+   * Get all existing SBT addresses
+   *
+   */
+  public async getAllSbtIds(): Promise<Array<string>> {
+    const data = await this.root.api.query.extendedAssets.soulboundAsset.entries();
+
+    return data.map(([code]) => code.args[0].toJSON().code) as Array<string>;
+  }
+
+  /**
+   * Checks whether provided asset is regulated to operate on it
+   * among KYC-verified accounts
+   * @param assetId asset ID
+   *
+   */
+  public async isAssetRegulated(assetId: string): Promise<boolean> {
+    return (await this.root.api.query.extendedAssets.regulatedAsset(assetId)).toJSON();
+  }
+
+  /**
+   * Get SBT expiration on account
+   * @param accountId account address
+   * @param sbtAssetId asset ID of SBT
+   *
+   */
+  public async getSbtExpiration(accountId: string, sbtAssetId: string): Promise<number> {
+    return (await this.root.api.query.extendedAssets.sbtExpiration(accountId, sbtAssetId)).unwrap().toNumber();
```

Comment:
unwrap might return unpredictable object

Reply:
lets use unwrapOr(null)?.toNumber() ?? 0

Reply:
0 is not an option, since you can expire someone's status by providing 0
to differentiate between those users I would write this and parse it based on response

unwrapOr('null').toString() - to know exactly if user was expired or unset at all